lib/package-doc.js

```diff
@@ -0,0 +1,44 @@
+var assign = require('lodash').assign
+var request = require('request')
+var resolve = require('url').resolve
+
+function PackageDoc (opts) {
+  assign(this, {
+    registry: process.env.FRONT_DOOR_HOST,
+    secret: process.env.SHARED_FETCH_SECRET
+  }, opts)
+}
+
+PackageDoc.prototype.get = function (pkgName, cb) {
+  pkgName = pkgName.replace('/', '%2f')
+
+  if (!this.registry) return cb(Error('no registry configured'))
+
+  request.get({
+    url: resolve(this.registry, pkgName),
+    json: true,
+    qs: {
+      sharedFetchSecret: this.secret
+    }
+  }, function (err, res, obj) {
+    if (res && res.statusCode >= 400) {
+      err = Error('unexpected status = ' + res.statusCode)
+      err.code = res.statusCode
+    }
+    if (err) {
+      return cb(err)
+    }
+
+    // only return latest version.
+    var latest
+    if (obj['dist-tags'] && obj['dist-tags'].latest && obj.versions) {
+      latest = obj.versions[obj['dist-tags'].latest]
```

Comment:
We should be, but it's a known issue in the registry; publish bugs years ago and we've never done data migration.

It would help cleanup code in various places if we eventually bit the bullet and did some data cleanup, but beyond the scope of this pull -- something we should slap ourselves on the wrist for every time we see this logic though!